### DISCUSSION TITLE
View not automatically updating when two trame apps running in same jupyter notebook

### DISCUSSION THREAD

**Discussion by cardinalgeo** [Kitware/trame#409] [Q&A] [answered]
Hi Trame team! 

When I run two trame apps within a jupyter notebook, only the last is automatically updated upon changing, e.g., the opacity of an actor. The first is updated only upon clicking within the viewer. In contrast, I'd expect each to update automatically, independent of the number of trame apps running. Any ideas on what I need to change to make this work? Here's a simple example that illustrates the issue: 

```python
# cell 1
from trame.app import get_server
from trame.ui.vuetify import SinglePageLayout
from trame.widgets import vuetify
import pyvista as pv

class Cone():
    def __init__(self, name):
        server = get_server(name)
        
        p = pv.Plotter()
        p.add_mesh(pv.Cone(), name="cone")
        self.plotter = p

        with SinglePageLayout(server) as layout:
            with layout.content:
                with vuetify.VContainer(fluid=True, classes="pa-0 fill-height"):
                    pv.trame.PyVistaRemoteView(p)

            self.server = server
            self.ui = layout

    def update_opacity(self, value): 
        self.plotter.actors["cone"].prop.opacity = value
        self.plotter.update()

# cell 2
cone_1 = Cone("1")
await cone_1.ui.ready
cone_1.ui

# cell 3
cone_2 = Cone("2")
await cone_2.ui.ready
cone_2.ui

# cell 4
cone_1.update_opacity(0.5) # cone opacity doesn't change without clicking 

# cell 5
cone_2.update_opacity(0.5) # cone opacity changes automatically
```

**Comment by jourdain** [comment, marked as answer]
Yes it is an issue with remote VTK and more than one server within the same interpreter. Feel free to create an issue in trame-vtk.

**Reply by jourdain** [reply]
using a child_server should work.

**Comment by cardinalgeo** [comment]
Thanks Seb! I'll give the `child_server` approach a try, open an issue in trame-vtk, and link it back here.

**Reply by cardinalgeo** [reply]
For those stumbling upon this, the original issue was resolved in issue [#60](https://github.com/Kitware/trame-vtk/issues/60) over in the `trame-vtk` repo without needing to use a `child-server`.
